raiwidgets/tests/test_cohort.py

```diff
@@ -141,7 +150,343 @@ def test_cohort_filter_serialization_include_exclude_methods(self, method):
         assert 'age' in json_str
 
 
+class TestCohortFilterDataValidations:
+    def test_validate_with_test_data_high_level_validations(self):
+        test_data = pd.DataFrame(data=[[23, 'X'], [25, 'Y']],
+                                 columns=["age", "target"])
+
+        cohort_filter_not_a_feature = \
+            CohortFilter(method=CohortFilterMethods.METHOD_LESS,
+                         arg=[65], column='fake_column')
+
+        with pytest.raises(
+                UserConfigValidationException,
+                match="Unknown column fake_column specified in cohort filter"):
+            cohort_filter_not_a_feature._validate_with_test_data(
+                test_data=test_data, target_column="target",
+                categorical_features=[])
+
+    def test_validate_with_test_data_index_filter_validations(self):
+        test_data = pd.DataFrame(data=[[23, 'X'], [25, 'Y']],
+                                 columns=["age", "target"])
+
+        cohort_filter_index_excludes = \
+            CohortFilter(method=CohortFilterMethods.METHOD_EXCLUDES,
+                         arg=[65], column=CohortFilter.INDEX)
+        with pytest.raises(
+                UserConfigValidationException,
+                match="excludes filter is not supported with Index based "
+                      "selection."):
+            cohort_filter_index_excludes._validate_with_test_data(
+                test_data=test_data, target_column="target",
+                categorical_features=[]
+            )
+
+        cohort_filter_index_incorrect_args = \
+            CohortFilter(method=CohortFilterMethods.METHOD_GREATER,
+                         arg=[65.0], column=CohortFilter.INDEX)
+        with pytest.raises(
+                UserConfigValidationException,
+                match="All entries in arg should be of type int."):
+            cohort_filter_index_incorrect_args._validate_with_test_data(
+                test_data=test_data, target_column="target",
+                categorical_features=[]
+            )
+
+    def test_validate_with_test_data_classification_error_filter_validations(
+            self):
+        test_data_multiclass = pd.DataFrame(
+            data=[[23, 'X'], [25, 'Y'], [25, 'Z']],
+            columns=["age", "target"])
+
+        test_data_binary = pd.DataFrame(
```

Comment:
I feel like these repeating test_data declarations should be refactored into some common method or utility

Reply:
done. Made function for these toy datasets